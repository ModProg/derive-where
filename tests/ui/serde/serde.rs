use std::marker::PhantomData;

use derive_where::derive_where;

#[derive_where(Clone)]
#[serde(crate = "serde_")]
struct InvalidSerde<T>(PhantomData<T>);

#[derive_where(Deserialize)]
#[serde(bound = "")]
struct ConflictingBound<T>(PhantomData<T>);

fn main() {}
