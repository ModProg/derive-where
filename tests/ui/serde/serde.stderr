error: Found unused `#[serde(...)]`
 --> tests/ui/serde/serde.rs:6:1
  |
6 | #[serde(crate = "serde_")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Found conflicting `#[serde(bound ...)]`
  --> tests/ui/serde/serde.rs:10:9
   |
10 | #[serde(bound = "")]
   |         ^^^^^^^^^^
